core/src/main/kotlin/fr/sncf/osrd/api/api_v2/stdcm/STDCMEndpointV2.kt

```diff
@@ -0,0 +1,172 @@
+    @Throws(OSRDError::class)
+    override fun act(req: Request): Response {
+        val recorder = DiagnosticRecorderImpl(false)
+        return try {
+            // Parse request input
+            val body = RqPrint(req).printBody()
+            val request =
+                stdcmRequestAdapter.fromJson(body)
+                    ?: return RsWithStatus(RsText("missing request body"), 400)
+
+            // parse input data
+            val infra = infraManager.getInfra(request.infra, request.expectedVersion, recorder)
+            val rollingStock = parseRawRollingStock(request.rollingStock)
+            val trainsRequirements =
+                parseRawTrainsRequirements(request.trainsRequirements, request.startTime)
+            val spacingRequirements = trainsRequirements.flatMap { it.spacingRequirements }
+
+            // Run the STDCM pathfinding
+            val path =
+                findPath(
+                    infra,
+                    rollingStock,
+                    request.comfort,
+                    0.0,
+                    parseSteps(infra, request.pathItems),
+                    makeBlockAvailability(
+                        infra,
+                        spacingRequirements,
+                        gridMarginBeforeTrain = request.timeGapBefore.seconds,
+                        gridMarginAfterTrain = request.timeGapAfter.seconds
+                    ),
+                    request.timeStep!!.seconds,
+                    request.maximumDepartureDelay!!.seconds,
+                    request.maximumRunTime.seconds,
+                    request.speedLimitTag,
+                    parseMarginValue(request.margin),
+                    Pathfinding.TIMEOUT
+                )
+            if (path == null) {
+                val error = OSRDError(ErrorType.PathfindingGenericError)
+                return ExceptionHandler.toResponse(error)
+            }
+            val pathfindingResponse = runPathfindingPostProcessing(infra, path.blocks)
+
+            // Run the simulation
+            val simulationResponse =
+                runStandaloneSimulation(
+                    infra,
+                    path.trainPath,
+                    path.chunkPath,
+                    infra.blockInfra.chunksToRoutes(infra.rawInfra, path.chunkPath.chunks),
+                    null,
+                    rollingStock,
+                    request.comfort,
+                    RJSAllowanceDistribution.MARECO,
+                    request.speedLimitTag,
+                    DistanceRangeMapImpl(),
+                    false,
+                    request.timeStep.seconds,
+                    parseSimulationScheduleItems(path.stopResults),
+                    0.0,
+                    RangeValues(emptyList(), listOf(request.margin))
+                )
+
+            // Check for conflicts
+            val departureTime =
+                request.startTime.plus(ofMillis((path.departureTime * 1000).toLong()))
```

Comment:
If the train requirements have been imported as an offset from the departure time, then we shouldn't add the start time here

Reply:
The `trainsRequirements` here are train requirements that have been parsed and offset to the specified stdcm's start time. When we run `findPath`, departure time is the offset from 0 (i.e. start time). So, to check for conflicts, i'd say we do need to reoffset it by departure time. Could be wrong though. Is there anything wrong in my reasoning?